src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ArrayEditor.cs

```
@@ -22,12 +20,12 @@ public ArrayEditor(Type type) : base(type)
     ///  Gets or sets the data type this collection contains.
     /// </summary>
     protected override Type CreateCollectionItemType()
-        => CollectionType?.GetElementType();
+        => CollectionType?.GetElementType()!;
```

Please add a comment, explaining in detail why `CollectionType?.GetElement()!` will never return null.

---

This can actually be null, but changing it seems to require changing the base class which does not return null.
@lonitra should I change the base class? Alternatively, I could return string.Empty here instead, but this would mean a breaking change.

---

The base class does not guarantee that CollectionType is not null, it should throw in the constructor when a null is passed in. Once we throw this exception, we can get rid of the question mark - 
 protected override Type CreateCollectionItemType() => CollectionType.GetElementType()!;
 
 And for GetElementType returning null, I would throw InvalidOprationException, it could happen only if this type is not an array

---

@Tanya-Solyanik I added a separate commit with the proposed changes, but this affects lots of tests. Are we sure that we want to apply this change? If yes, I will update the tests. If not, any other suggestions?

---

Yes, please update the tests.